pagopa/io-app
# feat: [#178212350] New profile structure

## Short description
Changed profile structure, added two new screens(ProfileDataScreen and SecurityScreen), enable/disable biometric recognition moved into SecurityScreen and deleted BiometricRecognitionScreen.

## How to test
Move into profile screen and see if the structure is equal to Figma [Click here](https://www.figma.com/file/zzRuoFxorMJmdkOBZH8WRG/%23178212350?node-id=0%3A1). Check also if enable/disable biometric recognition works properly after structural changes.

NOTE: It was possible to navigate into BiometricRecognitionScreen only if isFingerprintAvailable boolean was true. Inside BiometricRecognitionScreen there was some logic that did not work properly, if isFingerprintAvailable boolean was false it should have shown a button to navigate into phone settings to confirm biometric recognition permission to the app, as I have already written you could have entered in BiometricRecognitionScreen if isFingerprintAvailable was true so if you was already inside that screen it means that isFingerprintAvailable was true so the button to navigate into phone settings was always hidden.
The idea of the new logic is to confirm biometric recognition permission to the app(this functionality should only work on IOS because Android doesn't need that kind of permissions) by pressing this ListItemComponent:
<img width="300" alt="Screenshot 2021-06-03 at 15 04 34" src="https://user-images.githubusercontent.com/61834253/120649542-2b385400-c47d-11eb-88e7-c406e300dc38.png">
on press it should bring up an alert to confirm/cancel navigation to IOS settings.
I have already added the alert, unfortunately I can't add the logic to check if there is biometric recognition permission on IOS because I can't test it on a physical device, you should create a new task.
I also added some FIXME comments in SecurityScreen.

**Before:**
<img width="300" alt="Screenshot 2021-06-03 at 15 22 18" src="https://user-images.githubusercontent.com/61834253/120651910-9125db00-c47f-11eb-9bb3-4573f655d656.png">
<img width="300" alt="Screenshot 2021-06-03 at 15 26 53" src="https://user-images.githubusercontent.com/61834253/120652617-38a30d80-c480-11eb-931d-4142bd6cf8f6.png">
<img width="300" alt="Screenshot 2021-06-03 at 15 28 53" src="https://user-images.githubusercontent.com/61834253/120652868-756f0480-c480-11eb-91e5-17fb1d53c066.png">

**After:**
<img width="300" alt="Screenshot 2021-06-03 at 15 32 33" src="https://user-images.githubusercontent.com/61834253/120653576-1bbb0a00-c481-11eb-9bc6-7c39f53fb531.png">
<img width="300" alt="Screenshot 2021-06-03 at 15 33 11" src="https://user-images.githubusercontent.com/61834253/120653599-22e21800-c481-11eb-9137-95e259dc6aff.png">
<img width="300" alt="Screenshot 2021-06-03 at 15 34 26" src="https://user-images.githubusercontent.com/61834253/120653721-41481380-c481-11eb-9e78-2f07c9d3cd77.png">
<img width="300" alt="Screenshot 2021-06-03 at 15 36 21" src="https://user-images.githubusercontent.com/61834253/120654013-8e2bea00-c481-11eb-8224-cc32bf882c19.png">
